pkg/querier/blocks_store_queryable.go

```diff
@@ -717,7 +718,9 @@ func (q *blocksStoreQuerier) fetchSeriesFromStores(ctx context.Context, sp *stor
 				return nil
```

Comment:
Can we use `gCtx` in the call to `c.Series`? Won't it be cancelled once `g.Wait()` returns and therefore we won't be able to stream any series? The docs for `WithContext` say:

> The derived Context is canceled the first time a function passed to Go returns a non-nil error or the first time Wait returns, whichever occurs first.

Reply:
Good point. @dimitarvdimitrov should we use the `reqCtx` for the call to `c.Series`? The requirement here is that the context should still be open until we have finished the query and not just the end of this function call, because with streaming we will now fetching the chunks later. I am not sure the implications of changing contexts here, maybe we can selectively use `gCtx` or `reqCtx` depending on if we are streaming or not.

Reply:
nice spot @charleskorn, looking at how `reqCtx` and `ctx` are created I think they will be cancelled shortly after `fetchSeriesFromStores` returns, which happens even before consuming any of the labels in the stream. I think we should use `q.ctx` instead

Reply:
Thanks, using q.ctx

Reply:
Faced an issue here. Using `q.ctx` makes many integration tests fails. And using `gCtx` make the streaming integration test fail for the reason we are trying to solve (context closed error). I will dig into it a bit more tomorrow.